util.go

```diff
@@ -181,3 +183,174 @@ func deleteOnPull(ctx context.Context, sourceFiles []string, local string) error
 
 	return catcher.Resolve()
 }
+
+// The archive/unarchive functions below are modified version of the same
+// functions from github.com/mholt/archiver.
+
+func tarFile(tarWriter *tar.Writer, dir, relPath string) error {
```

Comment:
should the utility stuff end up in the utility package?

Reply:
I don't have strong feelings about it going here or in utility, but I feel like there's not a strong use case for this function. Most use cases could probably be satisfied by the functionality in archiver. I just happened to need to slightly tweak archiver's algorithm to make it work more nicely with pail buckets.

Reply:
got it.